torch_geometric/data/hetero_data.py

```diff
@@ -673,6 +674,20 @@ def _remove_tensor(self, attr: TensorAttr) -> bool:
             return True
         return False
 
+    def _get_tensor_size(self, attr: TensorAttr) -> List[int]:
+        if not attr.is_set('index'):
+            attr.index = None
+        tensor = self._get_tensor(attr)
+        return list(tensor.size()) if tensor is not None else []
```

Comment:
Let's error out here in case the `attr` does not exist. I think `_get_tensor` should error out as well.